File: mad_question4/app/src/main/java/com/example/ques4_mad/MainActivity.java
package com.example.ques4_mad;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    Button btnGoogleSignIn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnGoogleSignIn = findViewById(R.id.btnGoogleSignIn);

        btnGoogleSignIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Simulate successful login and navigate to HomeActivity
                Intent intent = new Intent(MainActivity.this, HomeActivity.class);
                startActivity(intent);
                finish(); // Optional: so user can't go back with back button
            }
        });
    }
}

File: mad_question5/app/src/main/java/com/example/mad_question5/MainActivity.kt
package com.example.mad_question5

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.mad_question5.databinding.ActivityMainBinding
import android.content.Intent

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val myData = MyData(id = 1, name = "John Doe")
        val intent = Intent(this, SecondActivity::class.java)
        intent.putExtra("myData", myData)
        startActivity(intent)

        binding.textView.text = "Hello, World!"
    }
}
